### PR TITLE
 #161499257 setup database

### PR DESCRIPTION
**Pull request by JoshuaKodhe**
#### What does this PR do?

Create a database connection as well as implementing the 
first authentication endpoint that is register

#### Description of Task to be completed?

Refactoring the folder structure to have the v2 application only
Installing Postgres and making a connection to the database using psycopg2
Writing tests for the register user endpoint as well as the logic for the endpoint as well

### How should this be manually tested?

Clone the application into your local repository using git clone
Setup a virtual environment 
Install the requirements using - pip install -r requirements
Export the variables in the environment

#### What are the relevant pivotal tracker stories?
[161499257](https://www.pivotaltracker.com/n/projects/2202996l)

### CODE REVIEW

File: app/__init__.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+from instance.config import APP_CONFIG
+from app.v2.models.db_setup import DB
+
+JWT = JWTManager()
+
+
+def create_app(config_name):
+    """ Registering app confingurations """
+    app = Flask(__name__, instance_relative_config=True)
```
Comment (Bonifase) [inline app/__init__.py:13]: Why do you pass `instance_relative_config=True` in Flask function?

File: app/v2/models/user.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
+"""User class"""
+from datetime import datetime
+from passlib.hash import pbkdf2_sha256 as sha256
+
+
+from app.v2.utils.db_connection import init_db
```
Comment (Bonifase) [inline app/v2/models/user.py:6]: I told you to get rid of this folder called V2

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+atomicwrites==1.2.1
+attrs==18.2.0
+certifi==2018.10.15
+chardet==3.0.4
+Click==7.0
+coverage==4.5.1
+coveralls==1.5.1
+docopt==0.6.2
+Flask==1.0.2
+Flask-JWT-Extended==3.13.1
```
Comment (Bonifase) [inline requirements.txt:12]: Why have you installed two JWT in your requirements file?

File: tests/test_products.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
+import unittest
+from flask import json
+from app import create_app
+from .base_test import BaseTest, base_url
+
+
+class TestProducts(BaseTest):
+    def test_for_successful_product_registration(self):
```
Comment (Bonifase) [inline tests/test_products.py:8]: Users should register and login before posting a product, I don't see this happening in your tests

File: requirements.txt [hunk 5]
```diff
@@ -0,0 +1,38 @@
+aniso8601==3.0.2
+astroid==2.0.4
+atomicwrites==1.2.1
+attrs==18.2.0
+certifi==2018.10.15
+chardet==3.0.4
+Click==7.0
+coverage==4.5.1
+coveralls==1.5.1
+docopt==0.6.2
```
Comment (Bonifase) [inline requirements.txt:10]: Why do you have this package?

File: requirements.txt [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+Flask==1.0.2
+Flask-JWT-Extended==3.13.1
+Flask-RESTful==0.3.6
+gunicorn==19.9.0
+idna==2.7
+isort==4.3.4
+itsdangerous==0.24
+Jinja2==2.10
```
Comment (Bonifase) [inline requirements.txt:18]: What do you use this package for?

**Review by Bonifase — COMMENTED** [review]
Good job, see the comments left

File: requirements.txt (hunk 3 above)
Comment (JoshuaKodhe) [reply requirements.txt]: After installing Flask-JWT-Extended and pip freezing I noticed I had both in my environment.
I am going to remove the one not in use

Comment (JoshuaKodhe) [reply requirements.txt:12]: > Why have you installed two JWT in your requirements file?

After installing Flask-JWT-Extended and pip freezing I noticed I had both in my environment.
I am going to remove the one not in use

File: tests/test_products.py (hunk 4 above)
Comment (JoshuaKodhe) [reply tests/test_products.py:8]: > Users should register and login before posting a product, I don't see this happening in your tests

I am working on the login feature before I can be able to add the decorator to require login, you asked that I work on authentication before I focused on the other endpoints

File: requirements.txt (hunk 6 above)
Comment (JoshuaKodhe) [reply requirements.txt:18]: > What do you use this package for?

Jinja is a templating package installed along Flask, it is majorly used for rendering templates(HTML)
 itsdangerous is used for signing data, to ensure the integrity of data.

The other two packages appeared after I did a pip install, and I still do not know what they are for, but I am going to find out what they do

File: requirements.txt (hunk 5 above)
Comment (JoshuaKodhe) [reply requirements.txt:10]: Click is installed when one installs flask.
Coverage and coveralls allow me to get test coverage on my tests
I am not sure what docopt is but after I installed coverage, it was also installed

File: app/v2/models/user.py (hunk 2 above)
Comment (JoshuaKodhe) [reply app/v2/models/user.py:6]: > I told you to get rid of this folder called V2

Let me do that right away, I will, however, need help with defining my url_prefix, are there any resources you could recommend to help?

**Pull request merged**